### PR TITLE
Don't project through resize when computing vectorizable inputs and outputs

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Because resized inner most dim can not be vectorized.

Fixes https://github.com/NVIDIA/Fuser/issues/486

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
!build

File: csrc/scheduler/utils.cpp [hunk 1]
```diff
@@ -1088,10 +1088,7 @@ IterDomain* projectIdToRoot(
         }
       }
     } else if (expr->isA<Resize>()) {
-      auto resize = expr->as<Resize>();
-      if (resize->out() == projected_id) {
-        projected_id = resize->in();
-      }
+      projected_id = nullptr;
```
Comment (naoyam) [inline csrc/scheduler/utils.cpp:1091]: Shouldn't this only mater when `resize->out() == projected_id`?

Comment (zasdfgbnm) [reply csrc/scheduler/utils.cpp]: Oops, you are right. Fixed!

**Comment by zasdfgbnm** [comment]
!build

**Review by naoyam — APPROVED** [review]
LGTM

**Comment by zasdfgbnm** [comment]
Looks like there is some real failure after I resolved https://github.com/NVIDIA/Fuser/pull/506#discussion_r1237581594. Strange. Converting to draft.

File: csrc/scheduler/transpose.cpp [hunk 2]
```diff
@@ -62,7 +62,7 @@ class DomainMap : public pointwise_utils::DomainMap {
     const auto& root_dom = tv->getRootDomain();
     IterDomain* mapped_id = nullptr;
     for (auto i : c10::irange(root_dom.size())) {
-      if (ca_map_.idGraph().permissiveNodes().permissiveAreMapped(
+      if (ca_map_.idGraph().permissiveResizeNodes().permissiveAreMapped(
```
Comment (zasdfgbnm) [inline csrc/scheduler/transpose.cpp:65]: @naoyam We should use the most permissive mapping here. I think this is a bug unrelated to this change but being exposed for some reason. I don't know in detail why this was not causing any trouble before.

**Comment by zasdfgbnm** [comment]
!build

File: csrc/scheduler/transpose.cpp (hunk 2 above)
Comment (naoyam) [reply csrc/scheduler/transpose.cpp:65]: That seems to be the case. I'll check why it wasn't failing.

**Review by naoyam — APPROVED** [review]
LGTM

**Comment by zasdfgbnm** [comment]
!build

**Comment by csarofeen** [comment]
> Because resized inner most dim can not be vectorized.

@zasdfgbnm @naoyam why is this the case?

**Comment by zasdfgbnm** [comment]
We generally use consumer as reference tensors, and forward propagate transformations.
 And because of the resize, the innermost dim of the reference tensor is not exactly mapped to the fusion input. So If I split out a 4 for vectorization in the reference tensor, then this 4 is not guaranteed to correspond 4 contiguous items in the input. For example, you can have an input 1d tensor with size 4096, but you pad it to 1+4096+3=4100, the an inner 4 in this 4100 tensor might be a `(pad, value, value, value)` for the input. So we can not do vector 4 load.

**Pull request merged**